### ISSUE TITLE
Component clone tool should work on inspectors

### ISSUE THREAD

**Issue by art0007i** [Yellow-Dog-Man/Resonite-Issues#3154]
### Is your feature request related to a problem? Please describe.

Currently the component clone tool can clone a component from an inspector, but it cannot paste the component onto another inspector to apply the cloned components to the inspected object. Instead it will apply those components to the actual inspector itself.
I find it that most of the time objects I'd want to clone components onto do not have colliders, which means I can't use the component clone tool on it.
I am aware of the feature of "dragging" components between inspector, which can be nice, but the component clone tool does not require you to hold the component you want to clone all the time in your hand, and you can have multiple component clone tools to clone different components quickly. I think this tool could be very powerful and useful if it had this functionality.

### Describe the solution you'd like

clicking with the tool on an inspector would clone components onto the inspected object, instead of the panel itself.

### Describe alternatives you've considered

the drag drop functionality, but it has some limitations which the component clone does not have.

### Additional Context

this used to be the case on oldplatform but not sure why it was removed. best guess is that the new inspector doesn't allow you to click on it because it's a canvas which captures your clicks instead of letting the tool click the inspector, but I think that there is a better solution than removing the useful functionality.

**Comment by shiftyscales** [comment]
@art0007i - To use the component clone tool with the inspector- grab the slot you want to attach the cloned component to and press secondary. Does that not meet your needs in some way?

**Comment by 3x1t-5tyl3** [comment]
> [@art0007i](https://github.com/art0007i) - To use the component clone tool with the inspector- grab the slot you want to attach the cloned component to and press secondary. Does that not meet your needs in some way?

You used to be abel to grab the slot ref and hit secondary to make a template..

That's kinda impossible now as it is overriden by the copy command?
How are we supposed to clone *with it* so I can paste with it?
Also seperately: #3145

**Comment by art0007i** [comment]
@shiftyscales I see, I was not aware that pasting via slot reference even exists, I guess this doesn't meet my needs in the form that I've used a ToolSimulator to attach components to specific objects on the old platform, which is now not possible because the ToolSimulator cannot hold a reference to a slot. Though my use case might be going above the tools intended purpose at that point.

I still think that this behavior is slightly intuitive because both the copy and paste actions are on secondary, but change depending on what reference you are holding (slot or component)

**Comment by 3x1t-5tyl3** [comment]
> [@shiftyscales](https://github.com/shiftyscales) I see, I was not aware that pasting via slot reference even exists, I guess this doesn't meet my needs in the form that I've used a ToolSimulator to attach components to specific objects on the old platform, which is now not possible because the ToolSimulator cannot hold a reference to a slot. Though my use case might be going above the tools intended purpose at that point.
> 
> I still think that this behavior is slightly intuitive because both the copy and paste actions are on secondary, but change depending on what reference you are holding (slot or component)

Doesn't this mean that while the component clone supports multiple components, it is never able to do so alone? lol

**Comment by shiftyscales** [comment]
Curious for input from @ProbablePrime or @Frooxius.

> Doesn't this mean that while the component clone supports multiple components

I wasn't aware the component clone tip could ever officially support / clone multiple components / an entire slot as a template- are you certain that functionality existed prior? The only time I had ever done anything similar was by manually inspecting the tool with a DevTool and adding additional components to the template slot myself- which goes beyond the scope of its intended use.

> I've used a ToolSimulator to attach components to specific objects on the old platform, which is now not possible because the ToolSimulator cannot hold a reference to a slot. Though my use case might be going above the tools intended purpose at that point.

Yeah- that's probably also a bit beyond the intended bounds of the tool's design- that would probably be best resolved by #57. Do you have content that was broken by that change that you're only just now noticing? Or was it just a temporary workflow solution you employed that isn't in any finalized projects?

**Comment by 3x1t-5tyl3** [comment]
> Curious for input from [@ProbablePrime](https://github.com/ProbablePrime) or [@Frooxius](https://github.com/Frooxius).
> 
> > Doesn't this mean that while the component clone supports multiple components
> 
> I wasn't aware the component clone tip could ever officially support / clone multiple components / an entire slot as a template- are you certain that functionality existed prior? The only time I had ever done anything similar was by manually inspecting the tool with a DevTool and adding additional components to the template slot myself- which goes beyond the scope of its intended use.
> 
> > I've used a ToolSimulator to attach components to specific objects on the old platform, which is now not possible because the ToolSimulator cannot hold a reference to a slot. Though my use case might be going above the tools intended purpose at that point.
> 
> Yeah- that's probably also a bit beyond the intended bounds of the tool's design- that would probably be best resolved by [#57](https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/57). Do you have content that was broken by that change that you're only just now noticing? Or was it just a temporary workflow solution you employed that isn't in any finalized projects?

Yes I remember it vividly and even Turk made an entire tool at the time that uses that functionality (Histeria Tip, Still can be found on one of my tools)
I frequently used it when making UIX panels as it would make it an absolute breeze to add entire templates.

**Comment by Frooxius** [comment]
Yes, the tool has supported multiple components since its initial design.

**Comment by shiftyscales** [comment]
I guess that functionality was overlooked when updating how they function with inspectors then, @Frooxius? Any suggestions for how to best resolve this issue, then?

**Comment by art0007i** [comment]
> Do you have content that was broken by that change that you're only just now noticing?

Yes. I have a pretty simple tool that attaches a dynamic variable space called "Data" onto the world root slot. I use this tool so I don't have to lag my game by opening the root slot in an inspector panel, and the "Data" space can be useful for specific music reactivity system which require it to be present
